### PR TITLE
Added validation for create booking endpoints

### PR DESCRIPTION
**Pull request by AlexeyShok**
Added validation for create booking endpoints

### CODE REVIEW

File: Api/Services/Accommodations/Bookings/BookingExecution/Flows/BankCreditCardBookingFlow.cs [hunk 1]
```diff
@@ -57,6 +58,7 @@ public async Task<Result<string>> Register(AccommodationBookingRequest bookingRe
             var availableCreditCardCurrency = _creditCardOptions.AvailableCurrencies;
 
             var (_, isFailure, booking, error) = await GetCachedAvailability(bookingRequest)
+                .Ensure(IsBookingOnline, "Booking on request is not allowed")
```
Comment (oleg-n-k) [inline Api/Services/Accommodations/Bookings/BookingExecution/Flows/BankCreditCardBookingFlow.cs:61]: AreAllBookingsAvailableEmmediately ?

Comment (oleg-n-k) [reply Api/Services/Accommodations/Bookings/BookingExecution/Flows/BankCreditCardBookingFlow.cs:61]: something like this in a message "On request booking aren't allowed currently"

Comment (AlexeyShok) [reply Api/Services/Accommodations/Bookings/BookingExecution/Flows/BankCreditCardBookingFlow.cs]: Renamed

File: Api/Services/Accommodations/Bookings/BookingExecution/Flows/FinancialAccountBookingFlow.cs [hunk 2]
```diff
@@ -54,6 +55,7 @@ public async Task<Result<AccommodationBookingInfo>> BookByAccount(AccommodationB
                 return Result.Failure<AccommodationBookingInfo>(bookingAvailability.Error);
 
             return await bookingAvailability
+                .Ensure(IsBookingOnline, "Booking on request is not allowed")
```
[the remark this answers is not in the document]
Comment (AlexeyShok) [reply Api/Services/Accommodations/Bookings/BookingExecution/Flows/FinancialAccountBookingFlow.cs]: Renamed

**Review by oleg-n-k — APPROVED** [review]

**Pull request merged**